Use model id as tizen device name

bbrto21 opened the pull request:
Me and swift-kim came to the conclusion that this way is best without any upstream modification.

## After landing this PR
```bash
flutter-tizen devices
8 connected devices:

SM N950N (mobile)                • ce0717175871bf22017e • android-arm64  • Android 9 (API 28)
Linux (desktop)                  • linux                • linux-x64      • Linux
Web Server (web)                 • web-server           • web-javascript • Flutter Tools
Chrome (web)                     • chrome               • web-javascript • Google Chrome 87.0.4280.66
Tizen W-6.0-circle-x86 (mobile)  • emulator-26121       • flutter-tester • Tizen 6.0 (emulator)
Tizen T-samsung-5.5-x86 (mobile) • emulator-26111       • flutter-tester • Tizen 5.5 (emulator)
Tizen M-6.0-x86 (mobile)         • emulator-26101       • flutter-tester • Tizen 6.0 (emulator)
Tizen TM1 (mobile)               • 0000d84f00006200     • flutter-tester • Tizen 6.0

```
``` bash
flutter-tizen run
Multiple devices found:
Tizen W-6.0-circle-x86 (mobile)  • emulator-26121   • flutter-tester • Tizen 6.0 (emulator)
Tizen T-samsung-5.5-x86 (mobile) • emulator-26111   • flutter-tester • Tizen 5.5 (emulator)
Tizen M-6.0-x86 (mobile)         • emulator-26101   • flutter-tester • Tizen 6.0 (emulator)
Tizen TM1 (mobile)               • 0000d84f00006200 • flutter-tester • Tizen 6.0
[0]: Tizen W-6.0-circle-x86 (emulator-26121)
[1]: Tizen T-samsung-5.5-x86 (emulator-26111)
[2]: Tizen M-6.0-x86 (emulator-26101)
[3]: Tizen TM1 (0000d84f00006200)
Please choose one (To quit, press "q/Q"): 
```
Signed-off-by: Boram Bae <boram21.bae@samsung.com>

swift-kim commented:
This breaks #3. The name was intentionally hard-coded to smoothly work with `-d tizen`.

bbrto21 commented:
> This breaks #3. The name was intentionally hard-coded to smoothly work with `-d tizen`.

Oh, I didn't know that. But it's a bit uncomfortable, What exactly should we do without -d?
## as-is
```
flutter-tizen devices
8 connected devices:

SM N950N (mobile) • ce0717175871bf22017e • android-arm64  • Android 9 (API 28)
Linux (desktop)   • linux                • linux-x64      • Linux
Web Server (web)  • web-server           • web-javascript • Flutter Tools
Chrome (web)      • chrome               • web-javascript • Google Chrome 87.0.4280.66
Tizen (mobile)    • emulator-26121       • flutter-tester • Tizen 6.0 (emulator)
Tizen (mobile)    • emulator-26111       • flutter-tester • Tizen 5.5 (emulator)
Tizen (mobile)    • emulator-26101       • flutter-tester • Tizen 6.0 (emulator)
Tizen (mobile)    • 0000d84f00006200     • flutter-tester • Tizen 6.0
```
## after
```
flutter-tizen devices
8 connected devices:

SM N950N (mobile)          • ce0717175871bf22017e • android-arm64  • Android 9 (API 28)
Linux (desktop)            • linux                • linux-x64      • Linux
Web Server (web)           • web-server           • web-javascript • Flutter Tools
Chrome (web)               • chrome               • web-javascript • Google Chrome 87.0.4280.66
W-6.0-circle-x86 (mobile)  • emulator-26121       • flutter-tester • Tizen 6.0 (emulator)
T-samsung-5.5-x86 (mobile) • emulator-26111       • flutter-tester • Tizen 5.5 (emulator)
M-6.0-x86 (mobile)         • emulator-26101       • flutter-tester • Tizen 6.0 (emulator)
TM1 (mobile)               • 0000d84f00006200     • flutter-tester • Tizen 6.0
```

swift-kim commented:
I don't have any good workaround at the moment. You'll have to modify the upstream flutter_tools logic to achieve both.

If you're only concerned with the output of the `devices` command, you may change the `Tizen 6.0` part to something like below. I guess the `sdkNameAndVersion` property is being used for it right now.

```
flutter-tizen devices
8 connected devices:

SM N950N (mobile) • ce0717175871bf22017e • android-arm64  • Android 9 (API 28)
Linux (desktop)   • linux                • linux-x64      • Linux
Web Server (web)  • web-server           • web-javascript • Flutter Tools
Chrome (web)      • chrome               • web-javascript • Google Chrome 87.0.4280.66
Tizen (mobile)    • emulator-26121       • flutter-tester • Tizen 6.0 (W-6.0-circle-x86) (emulator)
Tizen (mobile)    • emulator-26111       • flutter-tester • Tizen 5.5 (T-samsung-5.5-x86) (emulator)
Tizen (mobile)    • emulator-26101       • flutter-tester • Tizen 6.0 (M-6.0-x86) (emulator)
Tizen (mobile)    • 0000d84f00006200     • flutter-tester • Tizen 6.0 (TM1)
```

bbrto21 commented:
hm... I think that we can find the first device name using sdb and we can use this as an argument instead of 'tizen' in executable.dart.

swift-kim commented:
That would be an option. I tested with this code and at least it seemed to work:
```dart
  args = <String>[
    '--suppress-analytics', // Suppress flutter analytics by default.
    '--no-version-check',
    ...args,
  ];

  if (!args.contains('-d') && !args.contains('--device-id')) {
    await runInContext<void>(() async {
      final TizenDevices deviceDiscoverer = TizenDevices(
        logger: globals.logger,
        processManager: globals.processManager,
      );
      final List<Device> devices = await deviceDiscoverer.devices;
      if (devices.isNotEmpty) {
        args.insertAll(0, <String>['-d', devices.first.id]);
      }
    });
  }
```
But still it's very inefficient since everything is initialized twice in the AppContext (due to the use of `runInContext`). I'm not sure if there is any workaround.

bbrto21 commented:
We don't have to initialize twice. but I think that it looks better than using all the device names as 'Tizen'.
Of course, below code block is also a kind of workaround.
```dart
  String firtDeviceID;
  if (!specifiedId) {
    final String out = Process.runSync('sdb', ['devices']).stdout.toString();
    for (final String line in LineSplitter.split(out)) {
      if (line.startsWith('* ') ||
          line.startsWith('  ') ||
          line.startsWith('List of devices')) {
        continue;
      }

      final List<String> splitLine = line.split(RegExp(r'\s{2,}'));
      if (splitLine.length != 3) {
        continue;
      }

      firtDeviceID = splitLine[0];
      break;
    }
  }

  args = <String>[
    '--suppress-analytics', // Suppress flutter analytics by default.
    '--no-version-check',
    if (!specifiedId) ...<String>['-d', firtDeviceID],
    ...args,
  ];

```

swift-kim reviewed (approved):
LGTM
